app/ui-react/syndesis/src/modules/connections/pages/create/ConfigurationPage.tsx

```diff
@@ -97,8 +97,10 @@ export const ConfigurationPage: React.FunctionComponent = () => {
    * the API in the document. This cookie will be later used by the BE in the
    * redirect page set up in the 3rd party.
    */
-  if (connectResource) {
+  if (connectResource && connectResource.state) {
```

Comment:
Oh I see the logic but the fix is not the mutation (never mutate an object in js), but rather wrapping this code in an useEffect to avoid it run on subsequent renders. I can do this in the next few if you want!